Tasks/GooglePlayReleaseV4/GooglePlay.ts

```diff
@@ -0,0 +1,489 @@
+        const userFraction: number = Number(userFractionSupplied ? tl.getInput('userFraction', false) : 1.0);
+
+        const uploadMappingFile: boolean = tl.getBoolInput('shouldUploadMappingFile', false) && (action === 'SingleApk' || action === 'SingleBundle');
+        const mappingFilePattern: string = tl.getInput('mappingFilePath');
+
+        const changesNotSentForReview: boolean = tl.getBoolInput('changesNotSentForReview');
+
+        const releaseName: string = tl.getInput('releaseName', false);
+
+        const versionCodeFilterType: string = tl.getInput('versionCodeFilterType', false) || 'all';
+        let versionCodeFilter: string | number[] = null;
+        if (versionCodeFilterType === 'list') {
+            versionCodeFilter = getVersionCodeListInput();
+        } else if (versionCodeFilterType === 'expression') {
+            versionCodeFilter = tl.getInput('replaceExpression', true);
+        }
+
+        // Warn about unused inputs
+
+        switch (action) {
+            case 'MultiApkAab': warnIfUnusedInputsSet('bundleFile', 'apkFile', 'shouldUploadMappingFile', 'mappingFilePath'); break;
+            case 'SingleBundle': warnIfUnusedInputsSet('apkFile', 'bundleFiles', 'apkFiles'); break;
+            case 'SingleApk': warnIfUnusedInputsSet('bundleFile', 'bundleFiles', 'apkFiles'); break;
+            case 'OnlyStoreListing': warnIfUnusedInputsSet('bundleFile', 'apkFile', 'bundleFiles', 'apkFiles', 'track'); break;
+        }
+
+        // The regular submission process is composed
+        // of a transction with the following steps:
+        // -----------------------------------------
+        // #1) Get an OAuth token by authenticating the service account
+        // #2) Create a new editing transaction
+        // #3) Upload the new APK(s) or AAB(s)
+        // #4) Specify the track that should be used for the new APK/AAB (e.g. alpha, beta)
+        // #5) Specify the new change log
+        // #6) Commit the edit transaction
+
+        const globalParams: googleapis.Common.GlobalOptions = { auth: null, params: {} };
+
+        googleutil.updateGlobalParams(globalParams, 'packageName', packageName);
+
+        tl.debug('Initializing JWT.');
+        const jwtClient: googleapis.Common.JWT = googleutil.getJWT(key);
+        globalParams.auth = jwtClient;
+
+        tl.debug('Initializing Google Play publisher API.');
+        const edits: pub3.Resource$Edits = googleutil.publisher.edits;
+
+        tl.debug('Authorize JWT.');
+        await jwtClient.authorize();
+
+        console.log(tl.loc('GetNewEditAfterAuth'));
+        tl.debug('Creating a new edit transaction in Google Play.');
+        const edit: pub3.Schema$AppEdit = await googleutil.getNewEdit(edits, packageName);
+        googleutil.updateGlobalParams(globalParams, 'editId', edit.id);
+
+        let requireTrackUpdate = false;
+        const versionCodes: number[] = [];
+
+        if (action === 'OnlyStoreListing') {
```

Comment:
Just an idea for future improvements: Considering that now the logic of the pipeline task is dependent on a selected action. Could it make sense to encapsulate all needed logic inside action itself? We even could organize a factory for actions. The benefits of this approach are following:
1) We will decrease the amount of if-else statements
2) We could modify action and worry about impacting other actions
3) It will be easier to validate tasks inputs